### ISSUE TITLE
aria-valuenow, aria-valuemin, and aria-valuemax falsely reported as invalid on role="separator" element

### ISSUE THREAD

**Issue by hadleyel** [validator/validator#1507]
I think the validator is falsely reporting that the `aria-valuenow`, `aria-valuemin`, and `aria-valuemax` attributes are not allowed on `role="separator"` elements. I think the following code should be allowed, but it's possible I am misunderstanding part of the spec or just missing something!

### Steps to reproduce

Given the following code:
```html
<div
  role="separator"
  aria-label="Pane splitter"
  tabindex="0"
  aria-valuenow="51"
  aria-valuemin="1"
  aria-valuemax="101"
></div>
```

### Expected behavior

The validator should not give any errors for this code because it is valid according to the ARIA and ARIA in HTML specifications.

The [ARIA spec for `separator`](https://www.w3.org/TR/wai-aria/#separator) describes `aria-valuenow`, `aria-valuemin`, and `aria-valuemax` as necessary in some situations:
> If the `separator` is focusable, authors ***MUST*** set the value of [`aria-valuenow`](https://www.w3.org/TR/wai-aria/#aria-valuenow) to a [number](https://www.w3.org/TR/wai-aria/#valuetype_number) reflecting the current position of the `separator` and update that value when it changes. Authors ***SHOULD*** also provide the value of [`aria-valuemin`](https://www.w3.org/TR/wai-aria/#aria-valuemin) if it is not `0` and the value of [`aria-valuemax`](https://www.w3.org/TR/wai-aria/#aria-valuemax) if it is not `100`.

This paragraph is also unaltered in the [latest editor's draft](https://w3c.github.io/aria/#separator). No required context roles are listed.

Per [ARIA in HTML](https://www.w3.org/TR/html-aria/#el-div), it seems like `div` is allowed to have a `separator` role:
> [**Any `role`**](https://www.w3.org/TR/html-aria/#dfn-any-role) unless a direct child of a [`dl`](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element) element. Then only [`presentation`](https://www.w3.org/TR/html-aria/#index-aria-presentation) or [`none`](https://www.w3.org/TR/html-aria/#index-aria-none).
> [Naming Prohibited](https://www.w3.org/TR/html-aria/#dfn-naming-prohibited)
> Otherwise, [global `aria-*` attributes](https://www.w3.org/TR/wai-aria-1.2/#global_states) and any `aria-*` attributes applicable to the allowed roles.

### Actual behavior

The validator reports the following errors:

> Attribute `aria-valuenow` not allowed on element [`div`](https://html.spec.whatwg.org/multipage/#the-div-element) at this point.

> Attribute `aria-valuemin` not allowed on element [`div`](https://html.spec.whatwg.org/multipage/#the-div-element) at this point.

> Attribute `aria-valuemax` not allowed on element [`div`](https://html.spec.whatwg.org/multipage/#the-div-element) at this point.

**Comment by scottaohara** [comment]
agreed, these attributes, along with `aria-valuetext` should not be flagged so long as the `separator` also has a `tabindex`
